src/js/components/projects/createForm.tsx

```diff
@@ -29,40 +28,43 @@ interface ComboboxOption {
 interface Props extends RouteComponentProps {
   user: User;
   repository: Repository;
-  hasProjects: boolean;
+  isOpen: boolean;
+  closeCreateModal: () => void;
 }
 
-const ProjectForm = ({ user, repository, hasProjects, history }: Props) => {
+const CreateProjectModal = ({
+  user,
+  repository,
+  isOpen,
+  history,
+  closeCreateModal,
+}: Props) => {
   const isMounted = useIsMounted();
-  const [isOpen, setIsOpen] = useState(!hasProjects);
   // state related to setting base branch on project creation
   const [fromBranchChecked, setFromBranchChecked] = useState(false);
   const [fetchingBranches, setFetchingBranches] = useState(false);
   const [repoBranches, setRepoBranches] = useState<string[]>([]);
   const [filterVal, setFilterVal] = useState('');
-  const dispatch = useDispatch<ThunkDispatch>();
+  const [isSaving, setIsSaving] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const successTimeout = useRef<NodeJS.Timeout | null>(null);
 
-  const submitClicked = (e: React.MouseEvent<HTMLButtonElement>) => {
-    if (!isOpen) {
-      setIsOpen(true);
-      e.preventDefault();
+  const clearSuccessTimeout = () => {
```

Comment:
Depending on whether we show/hide this success text with CSS, we may want to consider making this `successTimeout` functionality into a hook since it's now being used in 3+ components;